### PR TITLE
Feature: Put the Tonberry Kinq on his throne

### PR DESCRIPTION
**Pull request by davismj**
- [x] Spawns on his thone
- [x] True lottery from Tonberry Dismayers in the throne room
- [x] Summons an elemental
- [x] Uses Astral Flow at 50% HP and summons an Avatar
- [x] Drops Tonberry Coat, Tonberry Lantern, Dried Mugwort, Water Spirit, Light Spirit
- [x] Drops 18000-30000 gil

image

### CODE REVIEW

File: sql/mob_spawn_points.sql [hunk 1]
```diff
@@ -46788,9 +46788,9 @@ INSERT INTO `mob_spawn_points` VALUES (17428673,'Tonberry_Dismayer','Tonberry Di
 INSERT INTO `mob_spawn_points` VALUES (17428674,'Tonberrys_Elemental','Tonberry\'s Elemental',8444,1,1,1,0);
 INSERT INTO `mob_spawn_points` VALUES (17428675,'Tonberry_Dismayer','Tonberry Dismayer',8436,-218,-0.792,24,127);
 INSERT INTO `mob_spawn_points` VALUES (17428676,'Tonberrys_Elemental','Tonberry\'s Elemental',8444,1,1,1,0);
-INSERT INTO `mob_spawn_points` VALUES (17428677,'Tonberry_Kinq','Tonberry Kinq',8438,0,0,0,0);
-INSERT INTO `mob_spawn_points` VALUES (17428678,'Tonberrys_Elemental','Tonberry\'s Elemental',8444,0,0,0,0);
-INSERT INTO `mob_spawn_points` VALUES (17428679,'Tonberrys_Avatar','Tonberry\'s Avatar',8443,0,0,0,0);
+INSERT INTO `mob_spawn_points` VALUES (17428677,'Tonberry_Kinq','Tonberry Kinq',8438,-219.9623,-1.7360,28.3876,64);
+INSERT INTO `mob_spawn_points` VALUES (17428678,'Tonberry_s_Elemental','Tonberry\'s Elemental',8444,1,1,1,0);
```
Comment (TeoTwawki) [inline sql/mob_spawn_points.sql]: Don't change the name here it was correct.

File: scripts/zones/Temple_of_Uggalepih/mobs/Tonberry_Dismayer.lua [hunk 2]
```diff
@@ -15,4 +16,5 @@ function onMobDeath(mob, player, isKiller)
     checkGoVregime(player,mob,793,1);
     checkGoVregime(player,mob,794,1);
     checkGoVregime(player,mob,795,1);
+    phOnDespawn(mob,TONBERRY_KINQ_PH,5,0); -- True lottery
```
[the remark this answers is not in the document]
Comment (TeoTwawki) [reply scripts/zones/Temple_of_Uggalepih/mobs/Tonberry_Dismayer.lua]: http://ffxiclopedia.wikia.com/wiki/Tonberry_Kinq?diff=prev&oldid=1209061
Going only by what the one anecdotal person from wiki says, that sounded more like a 6hr cooldown looking at his 2 edits to the page.

**Comment by davismj** [comment]
@TeoTwawki Appreciate the feedback! Changes made.

File: scripts/zones/Temple_of_Uggalepih/mobs/Tonberry_Dismayer.lua [hunk 3]
```diff
@@ -15,4 +16,5 @@ function onMobDeath(mob, player, isKiller)
     checkGoVregime(player,mob,793,1);
     checkGoVregime(player,mob,794,1);
     checkGoVregime(player,mob,795,1);
+    phOnDespawn(mob,TONBERRY_KINQ_PH,20,21600); -- 6 hours, 20% pop chance
```
Comment (TeoTwawki) [inline scripts/zones/Temple_of_Uggalepih/mobs/Tonberry_Dismayer.lua]: 20% rate is also pretty unlikely. You don't need to increase the rate to compensate for having a cooldown.

File: sql/mob_spawn_points.sql [hunk 4]
```diff
@@ -46788,9 +46788,9 @@ INSERT INTO `mob_spawn_points` VALUES (17428673,'Tonberry_Dismayer','Tonberry Di
 INSERT INTO `mob_spawn_points` VALUES (17428674,'Tonberrys_Elemental','Tonberry\'s Elemental',8444,1,1,1,0);
 INSERT INTO `mob_spawn_points` VALUES (17428675,'Tonberry_Dismayer','Tonberry Dismayer',8436,-218,-0.792,24,127);
 INSERT INTO `mob_spawn_points` VALUES (17428676,'Tonberrys_Elemental','Tonberry\'s Elemental',8444,1,1,1,0);
-INSERT INTO `mob_spawn_points` VALUES (17428677,'Tonberry_Kinq','Tonberry Kinq',8438,0,0,0,0);
-INSERT INTO `mob_spawn_points` VALUES (17428678,'Tonberrys_Elemental','Tonberry\'s Elemental',8444,0,0,0,0);
-INSERT INTO `mob_spawn_points` VALUES (17428679,'Tonberrys_Avatar','Tonberry\'s Avatar',8443,0,0,0,0);
+INSERT INTO `mob_spawn_points` VALUES (17428677,'Tonberry_Kinq','Tonberry Kinq',8438,-219.9623,-1.7360,28.3876,64);
+INSERT INTO `mob_spawn_points` VALUES (17428678,'Tonberrys_Elemental','Tonberry\'s Elemental',8444,1,1,1,0);
+INSERT INTO `mob_spawn_points` VALUES (17428679,'Tonberry_s_Avatar','Tonberry\'s Avatar',8443,1,1,1,0);
```
Comment (TeoTwawki) [inline sql/mob_spawn_points.sql]: needs to be `Tonberrys_Avatar`

**Pull request merged**